daemon/cmd/daemon_main.go

```diff
@@ -1584,7 +1582,13 @@ func registerDaemonHooks(lc fx.Lifecycle, shutdowner fx.Shutdowner) error {
 }
 
 // runDaemon runs the old unmodular part of the cilium-agent.
-func runDaemon(ctx context.Context, cleaner *daemonCleanup, shutdowner fx.Shutdowner) {
+func runDaemon(ctx context.Context, cleaner *daemonCleanup, shutdowner fx.Shutdowner, clientset k8sClient.Clientset) {
+	// Set the k8s clients provided by the K8s client cell. The global clients will be refactored out
+	// by later commits.
+	if clientset.IsEnabled() {
```

Comment:
Does `clientset` and `IsEnabled` need to be thread-safe?

Reply:
It was immutable and derived from the config earlier, but when integrating I realized that datapath-mode=lb-only wants to disable k8s support even when it has been configured so I needed to add `Disable()`. That's only meant to be called before the start hook runs and only from the invoke methods or constructors all of which runs sequentially. So all this to say, no it doesn't need to be thread-safe (if used as intended). I'll try to get rid of `Disable()` in the future though and make this a bit safer. For now I'll still add a check to `Disable()` that it's only called before start so we don't flip the enabled state after the sequential start is done.